packages/SwingSet/src/kernel/vatTranslator.js

```diff
@@ -59,7 +60,11 @@ function makeTranslateKernelDeliveryToVatDelivery(vatID, kernelKeeper) {
     if (msg.result) {
       insistKernelType('promise', msg.result);
       const p = kernelKeeper.getKernelPromise(msg.result);
-      p.state === 'unresolved' || Fail`result ${msg.result} already resolved`;
+      assert.equal(
+        p.state,
+        'unresolved',
+        `result ${msg.result} already resolved`,
+      );
```

Comment:
Isn't `msg.result` potentially expensive to serialize? The code being replaced handles that better.
```suggestion
      p.state === 'unresolved' || Fail`result ${msg.result} already resolved`;
```

Reply:
In theory, yes, but each time I try to use that pattern, I spend an hour fighting with TSC. In this case, it breaks the assertion on the very next line, that wants to look at `p.decider`, which only exists for unresolved promises:

```
$ tsc
src/kernel/vatTranslator.js:64:10 - error TS2339: Property 'decider' does not exist on type 'PromiseRecord'.
  Property 'decider' does not exist on type 'SettledPromiseRecord'.

64       !p.decider || Fail`result ${msg.result} already has decider ${p.decider}`;
            ~~~~~~~

src/kernel/vatTranslator.js:64:71 - error TS2339: Property 'decider' does not exist on type 'PromiseRecord'.
  Property 'decider' does not exist on type 'SettledPromiseRecord'.

64       !p.decider || Fail`result ${msg.result} already has decider ${p.decider}`;
                                                                         ~~~~~~~
```

I know this is probably TSC not being clever enough to realize that a previous ersatz assertion pattern imposes a constraint on the type, which should be enough to satisfy a following check. But *I'm* not clever enough to trick TSC into recognizing the right thing without using `assert()`, which is more readable anyways.

(this PR turns on types where there weren't any before, so we're monotonically improving the type-ed-ness of this file, but that's not the reason for the PR, so I'm disinclined to spend a lot of time on it)

If you can show me the trick to make it happy, I'll apply it.

Reply:
```
      if (p.state !== 'unresolved') {
        throw Fail`result ${msg.result} already resolved`;
      }
```
seems to satisfy it.